Preview

MichaelDao opened the pull request:
- Character count
- Text preview
- warning for limit 
- validation for announcements @

rockmanvnx6 commented:
@MichaelDao 
Just review my update, refactor if needed. Then merge your pull request.

MichaelDao commented:
I fixed the bug, the algorithm is definitely not accurate until all fields are filled out, but you cant submit the sms without filling in everything anyways.

The algorithm is completely accurate, crosschecking with copypaste from HTML will always be innacurate. 

I managed to fix the concurrency issue with selecting day. Whenever we picked a day it would always run the function before the input was registered to setState. So i added a function within a function to ensure that it ran in sequential order.

Also i refactored yah code, check this pull request @rockmanvnx6
